### ISSUE TITLE
Error in creating training dataset after refining outliers

### ISSUE THREAD

**Issue by anna-teruel** [DeepLabCut/DeepLabCut#1673]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

Hi folks, 
A partner of mine in the lab is facing a problem I never had before. He created a project in his laptop (Windows, Deeplabcut Version 2.1.10.4), trained for it inside a cluster (Deeplabcut Version 2.2.0.6) and then exctacted outliers and refined them in his laptop. Once finished, and still inside his laptop, he merged the dataset and created the training dataset. I always create training dataset inside the cluster, where the training will be done. That’s why I didn't know how to face it. I wrote this issue on Image.sc forum (https://forum.image.sc/t/unable-to-create-training-dataset-after-outlier-refinement/62856) and in the end of that talk they suggested me addressing the issue to the github repo. 

Creating a training dataset on a different computer where the training will be performed was obviously a mistake. Thus, we decided to update his DLC version to 2.2.0.6 (which is the one installed in our cluster), create a new project and copy all labeled folders to this new project. However, we keep getting an error when trying to create training dataset. This error only occurs with refined folders. If we delete refined labels and only use the original extracted and labeled frames, we are able to create the training dataset and eventually trained the network. We are on our way to extract new outlier frames and try the whole process again. However, it is a pity not to be able to save corrected data.

The error we keep getting when trying to create training dataset with refined labels is the following I attach in this post. 

On Image.sc forum they have suggested that the fact that the error has both front and back slashes is the only problem they can see. They say that to them, that says that the code is not properly building these paths when project are moved between windows and linux/mac. 2.2.0.5 and 2.2.0.6 have new strategies that were supposed to fix this, so this might be a bug that [@jeylau](https://forum.image.sc/u/jeylau) will need to fix.

What do you think about it? BIG THANKS
If any more information is needed, please let me know. 

### Operating System

operating system

### DeepLabCut version

dlc version

### DeepLabCut mode

single animal

### Device type

gpu computer cluster (https://bioinfo.cipf.es/ubb/cluster/), and cpu (windows) 

### Relevant log output

```shell
/opt/sci-soft/software/DeepLabCut/2.2.0.6-Anaconda3-GPU/lib/python3.8/site-packages/deeplabcut/generate_training_dataset/trainingsetmanipulation.py:717: FutureWarning: `windows2l$
  warnings.warn(
/opt/sci-soft/software/DeepLabCut/2.2.0.6-Anaconda3-GPU/lib/python3.8/site-packages/pandas/io/pytables.py:314: PerformanceWarning:
your performance may suffer as PyTables will pickle object types that it cannot
map directly to c-types [inferred_type->mixed,key->axis1] [items->None]

  f(store)
DLC loaded in light mode; you cannot use any GUI (labeling, relabeling and standalone GUI)
Traceback (most recent call last):
  File "deeplabcut_cluster.py", line 15, in <module>
    deeplabcut.create_training_dataset(config_path, augmenter_type = 'imgaug', windows2linux=True)
  File "/opt/sci-soft/software/DeepLabCut/2.2.0.6-Anaconda3-GPU/lib/python3.8/site-packages/deeplabcut/generate_training_dataset/trainingsetmanipulation.py", line 865, in create_$
    data, MatlabData = format_training_data(
  File "/opt/sci-soft/software/DeepLabCut/2.2.0.6-Anaconda3-GPU/lib/python3.8/site-packages/deeplabcut/generate_training_dataset/trainingsetmanipulation.py", line 631, in format_$
    img_shape = read_image_shape_fast(os.path.join(project_path, *filename))
  File "/opt/sci-soft/software/DeepLabCut/2.2.0.6-Anaconda3-GPU/lib/python3.8/site-packages/deeplabcut/generate_training_dataset/trainingsetmanipulation.py", line 613, in read_im$
    with Image.open(path) as img:
  File "/opt/sci-soft/software/DeepLabCut/2.2.0.6-Anaconda3-GPU/lib/python3.8/site-packages/PIL/Image.py", line 2975, in open
    fp = builtins.open(filename, "rb")
FileNotFoundError: [Errno 2] No such file or directory: '/home/vteruel/manel/conductas-socialobjeto-manel-2022-02-07/labeled-data\\M2161_hab1_1\\img18883.png'
```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by jeylau** [comment]
Hello @anna-teruel, you could edit `project_path` in the config.yaml file manually and things should work fine afterwards. From the image path I can tell that you're on windows, whereas the project path points to a Unix machine. This manual operation is typically not required (unless something wrong happens when parsing the config) as DeepLabCut automatically updates that field to match the current OS/project folder location.
As a side note, "creating a training dataset on a different computer where the training will be performed was obviously a mistake" isn't true anymore, as paths to data or images are now independent of your work station OS. I hope that helps! ☺️

**Comment by anna-teruel** [comment]
I am sorry to reopen this issue. But haven't solved the problem yet. It seems to happen when I try to load old projects into new versions of deeplabcut. In the end my colleague created a new project and started once again. He didn't try to change paths in the config file. However, now a different colleague has the same issue, and I think that, perhaps, has something to do with old and new versions. She took an old training and used it to analyze new videos from the same paradigm. She extracted outliers and refined them. When we save refined frames it seems that it is being saved, as shown in the picture: 

<img width="1135" alt="Captura de pantalla 2022-02-22 a les 9 43 26" src="https://user-images.githubusercontent.com/78961609/155095827-8cb4e120-a42f-4d84-bf2c-46feee01be24.png">


However, in our console we get following warning/error: 
`2022-02-22 09:42:04.084 python[637:5981] Warning: Expected min height of view: (<NSButton: 0x7fbaef1b17c0>) to be less than or equal to 30 but got a height of 32.000000. This error will be logged once per view in violation.
2022-02-22 09:42:17.466 python[637:5981] Warning: Expected min height of view: (<NSButton: 0x7fbb3a5998c0>) to be less than or equal to 30 but got a height of 32.000000. This error will be logged once per view in violation.
Checking labels if they are outside the image
2022-02-22 09:43:19.041 python[637:5981] Warning: Expected min height of view: (<NSButton: 0x7fbacef182b0>) to be less than or equal to 30 but got a height of 32.000000. This error will be logged once per view in violation.
2022-02-22 09:43:19.042 python[637:5981] Warning: Expected min height of view: (<NSButton: 0x7fbacef3bdb0>) to be less than or equal to 30 but got a height of 32.000000. This error will be logged once per view in violation.
2022-02-22 09:43:33.591 python[637:5981] Warning: Expected min height of view: (<NSButton: 0x7fbb2fa18430>) to be less than or equal to 30 but got a height of 32.000000. This error will be logged once per view in violation.`

I have checked CollectedData files created when frames are refined and seems to be wrong. Somewhy it divides the path into three columns: 

<img width="849" alt="Captura de pantalla 2022-02-22 a les 9 48 47" src="https://user-images.githubusercontent.com/78961609/155096341-5933a891-aedc-4c79-93b7-3ff0521c53ca.png">

I thought I could manually edit it, however I think (I might be wrong) that training dataset is created from the .h5 file, which I don't know how to edit. 

After merging dataset everything seems to be fine: 
`Merged data sets and updated refinement iteration to 4.
Now you can create a new training set for the expanded annotated images (use create_training_dataset).`

I have tried, after that, to check my labels, but it doesn't seem to work. Not even with old folders from my project. 

`2022-02-22 09:52:57.740 python[637:5981] Warning: Expected min height of view: (<NSButton: 0x7fbaef0dec80>) to be less than or equal to 30 but got a height of 32.000000. This error will be logged once per view in violation.
Creating images with labels by Camila.
Traceback (most recent call last):
  File "/opt/anaconda3/envs/DEEPLABCUT_M1/lib/python3.7/site-packages/deeplabcut/gui/label_frames.py", line 192, in check_labelF
    check_labels(self.config, visualizeindividuals=False)
  File "/opt/anaconda3/envs/DEEPLABCUT_M1/lib/python3.7/site-packages/deeplabcut/generate_training_dataset/trainingsetmanipulation.py", line 310, in check_labels
    color_by=color_by,
  File "/opt/anaconda3/envs/DEEPLABCUT_M1/lib/python3.7/site-packages/deeplabcut/utils/visualization.py", line 307, in make_labeled_images_from_dataframe
    h, w = ic[0].shape[:2]
  File "/opt/anaconda3/envs/DEEPLABCUT_M1/lib/python3.7/site-packages/skimage/io/collection.py", line 294, in __getitem__
    self.data[idx] = self.load_func(self.files[n], **kwargs)
  File "/opt/anaconda3/envs/DEEPLABCUT_M1/lib/python3.7/site-packages/skimage/io/_plugins/imageio_plugin.py", line 10, in imread
    return np.asarray(imageio_imread(*args, **kwargs))
  File "/opt/anaconda3/envs/DEEPLABCUT_M1/lib/python3.7/site-packages/imageio/core/functions.py", line 159, in imread
    with imopen(uri, "ri", plugin=format) as file:
  File "/opt/anaconda3/envs/DEEPLABCUT_M1/lib/python3.7/site-packages/imageio/core/imopen.py", line 223, in imopen
    raise err_type(err_msg)
ValueError: ImageIO does not generally support reading folders. Limited support may be available via specific plugins. Specify the plugin explicitly using the `plugin` kwarg, e.g. `plugin='DICOM'``

Finally, when I try to create training dataset, get the following error: 
`Traceback (most recent call last):
  File "/opt/anaconda3/envs/DEEPLABCUT_M1/lib/python3.7/site-packages/deeplabcut/gui/create_training_dataset.py", line 345, in create_training_dataset
    augmenter_type=self.aug_choice.GetValue(),
  File "/opt/anaconda3/envs/DEEPLABCUT_M1/lib/python3.7/site-packages/deeplabcut/generate_training_dataset/trainingsetmanipulation.py", line 866, in create_training_dataset
    Data, trainIndices, nbodyparts, project_path
  File "/opt/anaconda3/envs/DEEPLABCUT_M1/lib/python3.7/site-packages/deeplabcut/generate_training_dataset/trainingsetmanipulation.py", line 631, in format_training_data
    img_shape = read_image_shape_fast(os.path.join(project_path, *filename))
  File "/opt/anaconda3/envs/DEEPLABCUT_M1/lib/python3.7/site-packages/deeplabcut/generate_training_dataset/trainingsetmanipulation.py", line 613, in read_image_shape_fast
    with Image.open(path) as img:
  File "/opt/anaconda3/envs/DEEPLABCUT_M1/lib/python3.7/site-packages/PIL/Image.py", line 2953, in open
    fp = builtins.open(filename, "rb")
IsADirectoryError: [Errno 21] Is a directory: '/Users/annateruel/LPS-Camila-2021-12-27/labeled-data' `

If I check my training dataset folder iterartion-4 there's a CollectedData file created. However, the path seems to be missing: 

<img width="1161" alt="Captura de pantalla 2022-02-22 a les 10 06 12" src="https://user-images.githubusercontent.com/78961609/155099070-cadbf96c-f1d6-4fbf-9615-0b1aba584c3a.png">

I think that is happening because refined labels are not correctly saved, somehow. 

I don't know how to fix this, but I am willing to do so because we don't want to lose all labeled frames and start once again. 

Any help or tip is welcome and appreciated. THANK YOU

**Comment by jeylau** [comment]
No worries, you are not going to lose your annotated data 🙂
Indeed, the paths are now stored without path separators (in 3 separate columns), so the file you show is written properly (you're right though, it's ultimately the h5 file rather than the csv that matters). Old data format should automatically be converted to the newer MultiIndex with deeplabcut 2.2.0.6, but I'll check whether I can reproduce your error!

**Comment by anna-teruel** [comment]
Thank you very much @jeylau !
If you need me to incorporate more data or information, let me know.

**Comment by aristotelis-economides** [comment]
Hey @anna-teruel , I tried to reproduce your error using the [example project](https://github.com/DeepLabCut/DeepLabCut/tree/master/examples/openfield-Pranav-2018-10-30), which was created in an older deeplabcut version, therefore using the same label naming convention as your older projects. 

I trained a network using the existing labeled data, analyzed a video, extracted outlier frames, refined labels, and merged the data. Then I checked the labels and created a training dataset. Everything worked as expected, so either:
1. I did not follow your steps to reproduce the error correctly
2. The issue could be project specific in which case we could provide you with a small data conversion code snippet so that all your data are written correctly in the .h5 (and .csv) files

**Comment by anna-teruel** [comment]
Hi @intergalactic-mammoth , thanks a lot for trying to reproduce my error. Perhaps it would be more clear for you if you directly see our project folder. Is there a way I could send you our project folder from here? 

If you think it is easier, I can try with that small data conversion code snippet and see If we can save our labeled frames. 

Again, thanks for your help!

**Comment by MMathisLab** [comment]
hi @anna-teruel - I noticed when you import deeplabcut you have an issue with PyTables, which is important for creating H5 files. Maybe also try reinstalled DeepLabCut... 

```python
/opt/sci-soft/software/DeepLabCut/2.2.0.6-Anaconda3-GPU/lib/python3.8/site-packages/deeplabcut/generate_training_dataset/trainingsetmanipulation.py:717: FutureWarning: `windows2l$
  warnings.warn(
/opt/sci-soft/software/DeepLabCut/2.2.0.6-Anaconda3-GPU/lib/python3.8/site-packages/pandas/io/pytables.py:314: PerformanceWarning:
your performance may suffer as PyTables will pickle object types that it cannot
map directly to c-types [inferred_type->mixed,key->axis1] [items->None]

  f(store)
DLC loaded in light mode; you cannot use any GUI (labeling, relabeling and standalone GUI)
```

**Comment by aristotelis-economides** [comment]
> Perhaps it would be more clear for you if you directly see our project folder. Is there a way I could send you our project folder from here?

Hello @anna-teruel! You can also send us a Google Drive/ DropBox / link to your project and we can try to look into it

**Comment by jeylau** [comment]
@anna-teruel, the error has been fixed with #1763 and will be live in DeepLabCut 2.2.1 😃
